src/main/java/uk/gov/hmcts/ccd/auditlog/AuditInterceptor.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.ccd.auditlog;
+
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.lang.Nullable;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import uk.gov.hmcts.ccd.auditlog.aop.AuditAspect;
+import uk.gov.hmcts.ccd.auditlog.aop.AuditContext;
+import uk.gov.hmcts.ccd.auditlog.aop.AuditContextHolder;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+public class AuditInterceptor extends HandlerInterceptorAdapter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditAspect.class);
+
+    public static final String REQUEST_ID = "request-id";
+
+    private final AuditService auditService;
+
+    private static final List<HttpStatus> IGNORED_STATUSES = Lists.newArrayList(NOT_FOUND);
```

Comment:
this should be a config property rather than hardcoded?

Reply:
done